### PR TITLE
[ISDK-86] bugfix/refactor-login-registration-reset-forgot-password

### PR DESCRIPTION
**Pull request by DamengRandom**

### CODE REVIEW

File: src/pages/registration/register.page.ts [hunk 1]
```diff
@@ -1,87 +1,86 @@
 import { Component, ViewChild, OnInit, Inject } from '@angular/core';
 import { NgForm, FormBuilder, FormGroup, Validators } from '@angular/forms';
-import { NavController, ViewController, AlertController, LoadingController, NavParams } from 'ionic-angular';
+import { AlertController, LoadingController, NavController, NavParams, ViewController } from 'ionic-angular';
+import { loadingMessages, errMessages, generalVariableMessages } from '../../app/messages'; 
 import { Observable } from 'rxjs/Observable';
-
+import * as _ from 'lodash';
+import 'rxjs/add/operator/map';
```
Comment (imkven) [inline src/pages/registration/register.page.ts]: Do we need this?

Comment (DamengRandom) [reply src/pages/registration/register.page.ts]: Thx Kven, I will remove it.

File: src/app/messages.ts [hunk 2]
```diff
@@ -76,19 +76,22 @@ export const errMessages = {
...
+      "accepted": "You need to agree to the Terms & Conditions before completing registration."
     }
   },
   ResetPassword: {
     resetLoginFailed: {
       "failed": "Whoops, we were unable to reset your password. Please try again."
     },
     invalidLink: {
-      "invalid": "Oops... The log in link is broken. You will be redirected to the login page. Try again with your username and password."
+      "invalid": "Oops... The reset password link is expired, please contact your program coordinator."
```
Comment (jazzmind) [inline src/app/messages.ts]: "has" expired ... not "is"

File: src/pages/login/login.ts [hunk 3]
```diff
@@ -86,67 +86,47 @@ export class LoginPage {
...
-                        }
+              let getGame = this.gameService.getGames();
+              let getUser = this.authService.getUser();
+              let getMilestone = this.milestoneService.getMilestones();
+              Observable.forkJoin([getGame, getUser, getMilestone])
+                .subscribe(
+                  results => {
+                    loading.dismiss().then(() => {
+                      // results[0] game API data
+                      this.gameID = results[0].Games[0].id;
```
Comment (trtshen) [inline src/pages/login/login.ts]: @DamengRandom i believe you branched out from [ISDK-81](https://github.com/intersective/practera-sdk-ionic/pull/45), maybe you can start by looking into the fixing branch first.

Comment (DamengRandom) [reply src/pages/login/login.ts]: Yup sure @trtshen, I will merge my own branch first and make it consistent.

**Pull request merged**